src/main/java/ee/buerokratt/ruuter/domain/steps/SwitchStep.java

```diff
@@ -0,0 +1,33 @@
+package ee.buerokratt.ruuter.domain.steps;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import ee.buerokratt.ruuter.domain.Condition;
+import ee.buerokratt.ruuter.domain.ConfigurationInstance;
+import ee.buerokratt.ruuter.helper.ScriptingHelper;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import java.util.List;
+import java.util.Optional;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class SwitchStep extends ConfigurationStep {
+    @JsonAlias({"switch"})
+    private List<Condition> conditions;
+
+    @Override
+    public void execute(ConfigurationInstance configurationInstance) {
+        super.execute(configurationInstance);
+        ScriptingHelper scriptingHelper = configurationInstance.getScriptingHelper();
+        Optional<Condition> correctStatement = conditions.stream()
+            .filter(condition -> scriptingHelper.containsScript(condition.getConditionStatement())
```

Comment:
I think it's safe to leave out `scriptingHelper.containsScript(condition.getConditionStatement())` in this step. It was included in other "mapping steps", since in those there is a possibility that there either may or may not be a script present. Here however a script should always be present, so there's no point to first validate if it is or not.